### ISSUE TITLE
Unable to use $(tasks.status) in Pipeline

### ISSUE THREAD

**Issue by NithinKuruba** [tektoncd/pipeline#4514]
I am using Tekton pipeline in OCP platform. I am setting up notifications to convey the the overall status of the tasks. However, when I use $(tasks.status), I get below error.

`admission webhook "validation.webhook.pipeline.tekton.dev" denied the request: validation failed: invalid value: pipeline task status is not defined in the pipeline: spec.finally[1].params[aggregateTasksStatus].value`

**Comment by jerop** [comment]
@NithinKuruba I couldn't reproduce this issue - ran this example successfully:

https://github.com/tektoncd/pipeline/blob/4186dc96dfeea5bc465f94cf2c9cf359902973d2/examples/v1beta1/pipelineruns/pipelinerun-task-execution-status.yaml#L45-L58

Could you please share more details, such as:

- An example of the yaml, specifically the part that you're referring to the aggregate task status

- Kubernetes version:
  Output of `kubectl version`


- Tekton Pipeline version:
  Output of `tkn version` or `kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'`

My initial guess is that the reference to aggregate status is incorrect because the condition line 269 checking that the variable is an aggregate status evaluated to false, so it continued evaluating it as a reference to a specific pipeline task status, which could not be found and that's why you get the error from line 278

https://github.com/tektoncd/pipeline/blob/4186dc96dfeea5bc465f94cf2c9cf359902973d2/pkg/apis/pipeline/v1beta1/pipeline_validation.go#L264-L284

**Comment by NithinKuruba** [comment]
@jerop I am using openshift ver. 4.7 and the tekton pipelines ver. 1.4.1 provided by redhat. Below is the final task code I was testing with
```
- name: notifications
      params:
        - name: pipelineRunName
          value: $(context.pipelineRun.name)
      taskRef:
        kind: Task
        name: notifications-task
      when:
        - input: $(tasks.status)
          operator: in
          values:
            - Failed
            - None
```

**Comment by vdemeester** [comment]
@NithinKuruba given the version of tektoncd/pipeline that you are running (0.22.0) it is to be expected, as this variables (`tasks.status`) are only available since [v0.24.0](https://github.com/tektoncd/pipeline/releases/tag/v0.24.0). You would need to update your OpenShift Pipelines version to a more recent one (most likely by updating OpenShift to 4.8 that would bring you a more recent version of the tektoncd/pipeline shipped).

I will go ahead and close this issues as it is to be expected and is not really an issue.
/close
